### PR TITLE
Enhancement: Leverage LoggerAbstractServiceFactory for creating logger

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] adds a test to assert the service `ApplicationServiceErrorHandling` can be created
* [x] leverages `Zend\Log\LoggerAbstractServiceFactory` for creating logger
* [x] extracts a concrete factory for the `ApplicationServiceErrorHandling` service
* [x] leverages the `class` keyword as a service identifier
* [x] modifies type-hint to interface instead of concretion
* [x] reduces visibility of class property to `private`

### CODE REVIEW

File: module/Application/config/module.config.php [hunk 1]
```diff
@@ -85,12 +86,31 @@
...
     ],
+    'log' => [
+        'ZendLog' => [
+            'writers' => [
+                [
+                    'name' => 'stream',
+                    'options' => [
+                        'stream' => sprintf(
+                            './data/logs/log_%s.txt',
+                            date('F')
```
Comment (snapshotpl) [inline module/Application/config/module.config.php]: When config cache is enabled on production this value will stay constant

File: module/Application/config/module.config.php [hunk 2]
```diff
@@ -85,12 +86,31 @@
         ],
     ],
     'service_manager' => [
+        'abstract_factories' => [
+            Log\LoggerAbstractServiceFactory::class,
```
Comment (snapshotpl) [inline module/Application/config/module.config.php]: Why you don't used http://framework.zend.com/manual/current/en/modules/zend.mvc.services.html#zend-log-loggerabstractservicefactory ?

Comment (Ocramius) [inline module/Application/config/module.config.php]: @snapshotpl that is this factory? :)

File: module/Application/config/module.config.php (hunk 1 above)
Comment (localheinz) [inline module/Application/config/module.config.php]: @snapshotpl 

Good catch.

File: module/Application/config/module.config.php (hunk 2 above)
Comment (snapshotpl) [inline module/Application/config/module.config.php]: @snapshotpl go home, you are drunk :beers:

File: module/Application/config/module.config.php (hunk 1 above)
Comment (snapshotpl) [inline module/Application/config/module.config.php]: In monolog exists nice rotating file handler https://github.com/Seldaek/monolog/blob/master/src/Monolog/Handler/RotatingFileHandler.php
Maybe write something like that for Zend\Log?

Comment (Ocramius) [inline module/Application/config/module.config.php]: I also don't have anything against just using monolog, heh.

**Comment by localheinz** [comment]
@Ocramius @snapshotpl 

Working on integrating monolog, hang on!

**Comment by Ocramius** [comment]
@localheinz nice!

**Pull request closed without merging**